float32 x_dot
float32 y_dot
float32 z_dot